DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_increment_mac.py

```diff
@@ -0,0 +1,157 @@
+import asyncio
+import pytest
+import json
+import os
+
+from dent_os_testbed.lib.ip.ip_link import IpLink
+from dent_os_testbed.lib.onlp.onlp_system_info import OnlpSystemInfo
+from dent_os_testbed.constants import PLATFORMS_CONSTANTS
+
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen,
+    tgen_utils_setup_streams,
+    tgen_utils_start_traffic,
+    tgen_utils_stop_traffic,
+    tgen_utils_get_traffic_stats,
+    tgen_utils_get_loss,
+    tgen_utils_dev_groups_from_config,
+    tgen_utils_traffic_generator_connect,
+    tgen_utils_clear_traffic_items)
+from dent_os_testbed.utils.test_utils.br_utils import (
+    get_traffic_port_vlan_mapping,
+    configure_bridge_setup, 
+    configure_vlan_setup)
+
+
+pytestmark = [pytest.mark.suite_functional_vlan,
+              pytest.mark.asyncio,
+              pytest.mark.usefixtures("cleanup_tgen", "cleanup_bridges")]
+
+
+async def test_vlan_with_increment_macs(testbed):
+    """
+    Test Name: VLAN with increment mac
+    Test Suite: suite_functional_vlan
+    Test Overview: Test broadcast packet forwarding with incremented mac addresses
+    Test Procedure:
+    1. Create bridge. Set links, bridge to `up` state
+    2. Enslave interfaces to the created bridge entity.
+    3. Insert interfaces to VLAN
+        Port 1: vlan 5,
+                vlan 33,
+                vlan 20
+        Port 2: vlan 5
+                vlan 33,
+                vlan 20
+        Port 3: vlan 5,
+                vlan 33,
+                vlan 20
+        Port 4: vlan 5,
+                vlan 33,
+                vlan 20
+    4. Set up streams with incremented MAC address (16000 | 32000), pps 15000
+    5. Send traffic
+    6. Verify no packet loss nor packet leak occurred and all transmitted traffic received
+    7. Verify all MAC entries were learnt or re-learnt on the VLAN
+    8. Repeat steps #4 - #7 for vlans 5, 33, 20
+    """
+
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    device = dent_devices[0].host_name
+
+    tg_ports = tgen_dev.links_dict[device][0]
+    dut_ports = tgen_dev.links_dict[device][1]
+    tolerance = 0.7  # fdb learning tolerance
+    out = await OnlpSystemInfo.show(input_data=[{device: [{}]}], parse_output=True)
+    model = out[0][device]['parsed_output']['platform_name'].strip().replace("_", "-")
+
+    with open(PLATFORMS_CONSTANTS, "r") as f:
+        mac_table = f.read()
+    mac_count = json.loads(mac_table)
```

Comment:
I think that the test should not be opening files and parsing them. This should be done by a helper/utils method. Or you could make a fixture (maybe even with `session` scope) that will parse the config.

Reply:
i knew you will say that